bodhi/server/services/releases.py

```diff
@@ -325,6 +325,15 @@ def save_release(request):
             r = request.db.query(Release).filter(Release.name == edited).one()
             for k, v in data.items():
                 setattr(r, k, v)
+                # We have to change status for release updates to obsolete
+                if k == "state" and v == ReleaseState.archived:
```

Comment:
It might be good to make sure we only do this if the value is changing to archived, for efficiency.